### PR TITLE
docs(glossary): Add several web api entries in the glossary page

### PR DESCRIPTION
**Pull request by john015**
## Description

~~I added the mdn docs link to web apis(ex: localStorage, window, etc...) to make it easier for anyone who doesn't know about the web api~~
I added several web api entries in the glossary page

### Documentation

- https://www.gatsbyjs.org/docs/adding-react-components
- https://www.gatsbyjs.org/docs/debugging-html-builds
- https://www.gatsbyjs.org/docs/building-a-site-with-authentication
- https://www.gatsbyjs.org/docs/glossary
- https://www.gatsbyjs.org/docs/overview-of-the-gatsby-build-process
- https://www.gatsbyjs.org/docs/porting-from-create-react-app-to-gatsby
- https://www.gatsbyjs.org/docs/production-app
- https://www.gatsbyjs.org/docs/troubleshooting-common-errors
- https://www.gatsbyjs.org/docs/unit-testing

## Related Issues

 Related to #20678

### CODE REVIEW

**Comment by AishaBlake** [comment]
Thanks for taking this on, @john015! I'd love to see these explained better in the docs themselves. What do you think about either writing up some [glossary](https://www.gatsbyjs.org/docs/glossary/) entries for the APIs you've called out or opening an issue for the same?

**Comment by kingisaac95** [comment]
> Thanks for taking this on, @john015! I'd love to see these explained better in the docs themselves. What do you think about either writing up some [glossary](https://www.gatsbyjs.org/docs/glossary/) entries for the APIs you've called out or opening an issue for the same?

Quick first-time contributor question here. Do you mean getting a sentence or two of the explanations from the MDN page and adding it to the glossary?

**Comment by john015** [comment]
@AishaBlake, I agree with your opinion, so I added web api entries in the glossary page.

File: docs/docs/glossary.md [hunk 1]
```diff
@@ -192,6 +196,10 @@ JSX is an extension to JavaScript that allows developers to write HTML and custo
 
 Linting is the process of running a program that will analyze code for potential errors. The Gatsby project uses [prettier](https://prettier.io/) to identify and fix common style issues. Another example of a linter commonly used in React projects is [eslint-jsx-plugin-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y), which checks for common [accessibility](#accessibility) issues in development.
 
+### LocalStorage
+
+LocalStorage is the storage that to save key/value pairs in a web browser. The localStorage object stores data with no expiration date. The data will not be deleted when the browser is closed, and will be available the next day, week, or year.
```
Comment (AishaBlake) [inline docs/docs/glossary.md]: Same deal as with MDN. This is a little too similar to the definition on w3schools.

File: docs/docs/glossary.md [hunk 2]
```diff
@@ -204,6 +212,10 @@ A way of writing HTML content with plain text, using special characters to denot
 
 ## N
 
+### Navigator
+
+A Navigator object represents information about the browser. You can get the browser's language or user-agent using the Navigator object.
```
Comment (AishaBlake) [inline docs/docs/glossary.md]:
```suggestion
A Navigator object contains information about the browser. For example, you can get the browser's language via the `window.navigator.language` property.
```

I'm thinking a more specific example might be helpful here. Let me know what you think!

**Comment by AishaBlake** [comment]
> > Thanks for taking this on, @john015! I'd love to see these explained better in the docs themselves. What do you think about either writing up some [glossary](https://www.gatsbyjs.org/docs/glossary/) entries for the APIs you've called out or opening an issue for the same?
> 
> Quick first-time contributor question here. Do you mean getting a sentence or two of the explanations from the MDN page and adding it to the glossary?

Thanks for the question, @kingisaac95. I do not mean to say that anything should be copied from any other source and pasted into the Gatsby docs. If that _does_ happen, it should be done sparingly, thoughtfully, and with proper attribution. An exception, in my opinion, would be if the original author were making the contribution. I'd leave it up to them in that case!

**Comment by AishaBlake** [comment]
Closing this as #22069 has addressed the issue and this has gone more than a month without any activity, though I'd still love to see this information added to the glossary!

**Pull request closed without merging**